tsdb/compact.go

```diff
@@ -469,7 +478,7 @@ func (c *LeveledCompactor) CompactWithBlockPopulator(dest string, dirs []string,
 
 		if b == nil {
 			var err error
-			b, err = OpenBlock(c.logger, d, c.chunkPool)
+			b, err = OpenBlock(c.logger, d, c.chunkPool, c.postingsDecoder)
```

Comment:
Maybe it would be enough to pass `Block` to the decoder function so that we could dynamically use a different postings decoder?

Reply:
Then do we get the Thanos metadata using `Block`? I wonder if we can change the `tsdb.BlockMeta` struct of TSDB itself...
Actually https://github.com/prometheus/prometheus/blob/main/tsdb/block.go#L211 might be a place we can extend to contain postings format info without changing Thanos. WDYT

Reply:
Added the code to the latest commit